sshauthentication-api/src/main/java/org/openintents/ssh/authentication/util/key/SshEd25519PublicKey.java

```diff
@@ -65,17 +65,13 @@ public void setAByte(byte[] aByte) {
     // encode point according to RFC8032 section 5.1.5
     private byte[] encodePoint() {
         byte[] yIn = mY.toByteArray();
-        if (yIn.length > 32 && yIn[0] == 0) {
-            // BigInteger.toByteArray prepends a byte for sign bit if MSB of Y is set,
-            // but Y will always be positive, so remove the superfluous byte
-            yIn = Arrays.copyOfRange(yIn, 1, 33);
-        }
         byte[] y = new byte[32];
-        for (int i = 0; i < yIn.length; i++) {
-            y[i] = yIn[yIn.length-1 - i]; // convert to little endian
+        for (int i = 0; i < 32; i++) {
+            y[i] = yIn[31 - i]; // convert to little endian
```

Comment:
did you consider using ByteBuffer for this?

Reply:
Thanks. Regarding this whole function: Is it correct to take the LSB (x_0) of the big endian coordinate x? Comparing with other implementations this seems to be correct, but RFC8032 doesn't clearly specify it. It does mention in the decoding to set x = p-x if the encoded bit x_0 != x mod 2.

Reply:
I could remove the whole function since most users will probably use a crypto library that returns the already encoded point. It's mostly a product of investigating whether bouncycastle returns the correct format needed for SSH public key blobs.